crates/primitives/src/header.rs

```diff
@@ -176,8 +176,14 @@ impl Header {
     /// Calculate base fee for next block according to the EIP-1559 spec.
     ///
     /// Returns a `None` if no base fee is set, no EIP-1559 support
-    pub fn next_block_base_fee(&self) -> Option<u64> {
-        Some(calculate_next_block_base_fee(self.gas_used, self.gas_limit, self.base_fee_per_gas?))
+    pub fn next_block_base_fee(&self, change_denominator: u64, elasticity: u64) -> Option<u64> {
```

Comment:
this will be very inconvenient to use outside of the chainspec context.

I'd prefer if we could move the params into a new type with denominator and elasticity as fields so we have an easy way to get the right values via a function `ethereum()` for example.
otherwise callers always need to deal with the specific constants

Reply:
done